includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -3331,4 +3334,32 @@ public function is_payment_request_enabled() {
 		return in_array( WC_Stripe_Payment_Methods::APPLE_PAY, $enabled_payment_method_ids, true ) ||
 			in_array( WC_Stripe_Payment_Methods::GOOGLE_PAY, $enabled_payment_method_ids, true );
 	}
+
+	/**
+	 * Maybe deactivate Affirm or Klarna payment methods if other official plugins are active.
+	 *
+	 * @return void
+	 */
+	private function maybe_deactivate_bnpls() {
+		if ( WC_Stripe_Helper::has_other_bnpl_plugins_active() ) {
+			$enabled_payment_methods       = $this->get_upe_enabled_payment_method_ids();
+			$payment_method_ids_to_disable = [];
+			if ( in_array( WC_Stripe_Payment_Methods::AFFIRM, $enabled_payment_methods, true )
+				&& WC_Stripe_Helper::has_gateway_plugin_active( WC_Stripe_Helper::OFFICIAL_PLUGIN_ID_AFFIRM ) ) {
+				$payment_method_ids_to_disable[] = WC_Stripe_Payment_Methods::AFFIRM;
+			}
+			if ( in_array( WC_Stripe_Payment_Methods::KLARNA, $enabled_payment_methods, true )
+				&& WC_Stripe_Helper::has_gateway_plugin_active( WC_Stripe_Helper::OFFICIAL_PLUGIN_ID_KLARNA ) ) {
+				$payment_method_ids_to_disable[] = WC_Stripe_Payment_Methods::KLARNA;
+			}
+			if ( count( $payment_method_ids_to_disable ) > 0 ) {
```

Comment:
Non-blocking, but we could also use a simple comparison here.
```suggestion
			if ( $payment_method_ids_to_disable !== [] ) {
```